src/main/java/it/gov/pagopa/payment/controller/TransactionControllerImpl.java

```diff
@@ -0,0 +1,21 @@
+package it.gov.pagopa.payment.controller;
+
+import it.gov.pagopa.common.performancelogger.PerformanceLog;
+import it.gov.pagopa.payment.dto.TransactionDTO;
+import it.gov.pagopa.payment.service.TransactionService;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class TransactionControllerImpl implements TransactionController {
+    private final TransactionService transactionService;
+
+    public TransactionControllerImpl(TransactionService transactionService) {
+        this.transactionService = transactionService;
+    }
+
+    @Override
+    @PerformanceLog("GET_TRANSACTION")
+    public TransactionDTO getTransaction(String transactionId, String userId) {
```

Comment:
always log what's happening (if too verbose, we could use a lower level than INFO)
```suggestion
@RestController
@Slf4j
public class TransactionControllerImpl implements TransactionController {
    private final TransactionService transactionService;

    public TransactionControllerImpl(TransactionService transactionService) {
        this.transactionService = transactionService;
    }

    @Override
    @PerformanceLog("GET_TRANSACTION")
    public TransactionDTO getTransaction(String transactionId, String userId) {
       log.info("[GET_TRANSACTION] User {} requested to retrieve transaction {}", userId, transactionId)
```